src/reducer/async/actions.ts

```diff
@@ -97,12 +100,16 @@ function appendStateMetadata(
     appendedMetadata.description = state.fields.description;
   }
 
-  for (let row of state.metadataRows) {
+  for (let row of state.attributes) {
     if (row.name !== null && row.value !== null) {
-      appendedMetadata[row.name] = row.value;
+      if(!appendedMetadata.attributes) appendedMetadata.attributes = []
+      appendedMetadata.attributes.push({name: row.name, value: row.value});
     }
   }
 
+  if (!appendedMetadata.creators) appendedMetadata.creators = []
+  appendedMetadata.creators.push(system.tzPublicKey || "")
```

Comment:
We're going to add a `minter` field to TZIP-021 because the "creator(s)" are not necessarily the same. It will be a single scalar value with the spec `minter (string) [format: tzAddress]`. Can you please change to this?

Reply:
Sure.